src/mlpack/tests/main_tests/kmeans_test.cpp

```diff
@@ -72,7 +72,7 @@ BOOST_AUTO_TEST_CASE(PointsLessThanClustersTest)
   arma::mat inputData = arma::randu<arma::mat>(N, D);
 
   SetInputParam("input", std::move(inputData));
-  SetInputParam("clusters", (int) 11); // Invalid
+  SetInputParam("clusters", (int) 0); // Invalid
```

Comment:
Doesn't it break the idea of the test if you set the number of clusters to a value less than N?

Reply:
Actually, the idea here was changed because clusters demanded can be more than points according to existing implementation of `kmeans` So, here I'm testing that when 0 clusters is passed i.e. when clusters are to be auto detected then initial centroid must be passsed for autodetecting. I've changed the name of test now.

Reply:
Sure, I guess when I review the actual code, it is possible to ask for more clusters but in the resulting clusters it is guaranteed that some will be empty.